TextInputKit/TextInputKit/Code/SpecializedTextInput/CardNumber/BankCardNumberTextInputFormatter.swift

```diff
@@ -20,10 +20,88 @@ final class BankCardNumberTextInputFormatter : TextInputFormatter {
         replacing replacementString: String,
         at editedRange: Range<String.Index>) -> TextInputValidationResult {
 
-        // TODO: Implement.
-        return .accepted
+        let originalString = originalString.unicodeScalars
```

Comment:
It's expected to be string due to name, but is's unicode scalars representation. So any name like "originalUnicodeRep(resentation)"  may be used.
Here and below for all string representations.